backend/src/org/commcare/session/CommCareSession.java

```diff
@@ -612,31 +626,18 @@ private SessionFrame updateMatchingFrame(String frameId) {
         return null;
     }
 
-    private boolean pushNewFrame(SessionFrame matchingFrame, boolean newFrame, boolean currentFramePushed) {
-        // ok, frame should be appropriately modified now.
-        // we also need to push this frame if it's new
-        if (newFrame) {
-            // Before we can push a frame onto the stack, we need to
-            // make sure the stack is clean. This means that if the
-            // current frame has a snapshot, we've gotta make sure
-            // the existing frames are still valid.
-
-            // TODO: We might want to handle this differently in the future,
-            // so that we can account for the invalidated frames in the ui
-            // somehow.
-            cleanStack();
-
-            // OK, now we want to take the current frame and put it up on the frame stack unless
-            // this frame is dead (IE: We're closing it out). then we'll push the new frame
-            // on top of it.
-            if (!frame.isDead() && !currentFramePushed) {
-                frameStack.push(frame);
-                return true;
-            }
```

Comment:
Managing the "new Frame" code is important. If you are using ID's you can only have one frame with a given ID on the stack at a time. Removing this code will prevent that from working right.